prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -493,13 +494,33 @@ ssize_t rxr_pkt_post_read_to_copy_data(struct rxr_ep *rxr_ep,
 	assert(iov_idx >=0 && iov_idx < rx_entry->iov_count);
 	assert(efa_ep_is_cuda_mr(rx_entry->desc[iov_idx]));
 	assert(iov_offset + data_size <= rx_entry->iov[iov_idx].iov_len);
+	assert(!pkt_entry->next);
 
 	dest_buf = (char *)rx_entry->iov[iov_idx].iov_base + iov_offset;
 
-	pkt_entry->state = RXR_PKT_ENTRY_COPY_BY_READ;
-	assert(pkt_entry->mr);
+	if (pkt_entry->type == RXR_PKT_ENTRY_OOO || pkt_entry->type == RXR_PKT_ENTRY_UNEXP) {
```

Comment:
Perhaps check for no MR instead of the type? That's a bit cleaner way to check/works in all cases if something changes in the future.

Reply:
Sure, make sense.

Reply:
Updated.